lib/urlapi.c

```diff
@@ -251,13 +251,15 @@ bool Curl_is_absolute_url(const char *url, char *buf, size_t buflen)
       */
     }
     else {
-        break;
+      break;
     }
   }
   if(i && (url[i] == ':') && (url[i + 1] == '/')) {
     if(buf) {
-      memcpy(buf, url, i);
       buf[i] = 0;
+      while(i--) {
+        buf[i] = TOLOWER(url[i]);
```

Comment:
I think this needs the typecast from the original to avoid warnings:
~~~c
buf[i] = (char)TOLOWER(url[i]);